### PR TITLE
Incorporated functionality for pi-bots (non activitybot robots)

### PR DESCRIPTION
**Pull request by darrahts**
Need to incorporate the functionality for the raspberry pi based robots (python robots). Updated from previous pull request attempt, which I was working on an older version of roboscape.js. Tested and works.

### CODE REVIEW

**Comment by hamidzr** [comment]
- it looks like you support setting speeds for the robots. could you have it follow the same interface too? for example `set speed LEFTWHEEL RIGHTWHEEl` etc following the Roboscape docs that would make the these robots much more interchangeable with the activity bots
- Maybe we want to keep  the same range of valid speed values for the different robots

**Comment by hamidzr** [comment]
also please look at the codeclimate issues

**Comment by darrahts** [comment]
for the comments regarding lines 914 and 1013 I do not see anything wrong with these functions. The first function I didn't even write and was not part of my changes. 

```
    /**
     * Sets the client message limit and penalty for the given robot.
     * @param {string} robot name of the robot (matches at the end)
     * @param {number} rate number of messages per seconds
     * @param {number} penalty number seconds of penalty if rate is violated
     * @returns {boolean} True if the robot was found
     */
    RoboScape.prototype.setClientRate = function (robot, rate, penalty) {
        robot = this._getRobot(robot);
        if (robot && robot.accepts(this.socket.uuid)) {
            robot.setClientRate(rate, penalty);
            return true;
        }
        return false;
    };

    /**
     * stops the pi-bot robot
     * @param {string} robot name of the robot (matches at the end)
     */
    RoboScape.prototype.stop = function (robot)
    {
        robot = this._getRobot(robot);
        if(robot && robot.accepts(this.socket.uuid))
        {
            logger.log('stop');
            robot.stop();
            return true;
        }
        return false;
    };

```

File: src/server/rpc/procedures/roboscape/roboscape.js [hunk 1]
```diff
@@ -58,6 +69,93 @@ var Robot = function (mac_addr, ip4_addr, ip4_port) {
...
 };
 
+Robot.prototype.Quit = function () {
+    logger.log("sending quit signal to: " + this.mac_addr);
+    var message = Buffer.alloc(1);
+    message.write('Q', 0, 1);
+    this.sendToRobot(message);
+};
+
+Robot.prototype.Drive = function (dir, speed)
```
[the remark this answers is not in the document]
Comment (darrahts) [reply src/server/rpc/procedures/roboscape/roboscape.js]: fixed.

File: src/server/rpc/procedures/roboscape/roboscape.js [hunk 2]
```diff
@@ -22,7 +25,15 @@
  *  'D' left[2] right[2]: drive certain distance
  *  'L' led[1] state[1]: change LED state
  *  'G' msec[2] pwr[1]: send infra red light
- *
+ *  
+ * 'd' direction[1] speed[1] : set driving speed (python robots)
+ *  't' direction[1] angle[1] dimeTurn[1] (bool) : turn (python robots)
```
[the remark this answers is not in the document]
Comment (darrahts) [reply src/server/rpc/procedures/roboscape/roboscape.js]: 'D' and 'T' were already taken, but I still wanted to use letters that made sense with the meaning.

[the remark this answers is not in the document]
Comment (darrahts) [reply src/server/rpc/procedures/roboscape/roboscape.js]: why do my comments keep getting deleted? 'D' and 'T' were already taken.

[the remark this answers is not in the document]
Comment (hamidzr) [reply src/server/rpc/procedures/roboscape/roboscape.js]: > why do my comments keep getting deleted? 'D' and 'T' were already taken.

they all show up :)

**Comment by hamidzr** [comment]
yeah imo not all of the codeclimate issues were relevant.

File: src/server/rpc/procedures/roboscape/roboscape.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (darrahts) [reply src/server/rpc/procedures/roboscape/roboscape.js]: Ah there they are! :)

**Comment by darrahts** [comment]
Why does the continuous integration Travic CI build fail? There is nothing wrong with the new file...

**Comment by darrahts** [comment]
Going to extend the original robot class and put the additional functionalities in the new class.

**Comment by darrahts** [comment]
need to refactor pi-bot class out of robot class to reflect recent changes in the organization of roboscape. See #2357 .

**Pull request closed without merging**